Dockerfile

```diff
@@ -52,48 +52,46 @@ USER $USER_NAME
 # Build.
 RUN mvn package -Pproduction
 
-# Switch to Normal JRE Stage.
-FROM openjdk:11-jre-slim
+# JRE Stage.
+FROM eclipse-temurin:11-alpine
 ARG USER_ID
 ARG USER_NAME
+ARG HOME_DIR
 ARG SOURCE_DIR
+ARG APP_PATH
+
+ENV APP_PATH=$APP_PATH
+
+# Create the group (use a high ID to attempt to avoid conflits).
+RUN addgroup -g $USER_ID $USER_NAME
 
-RUN apt-get update && \
-    apt-get upgrade -y && \
-    apt-get -y install gettext-base && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
+# Create the user (use a high ID to attempt to avoid conflits).
+RUN adduser -h $HOME_DIR -u $USER_ID -G $USER_NAME -D $USER_NAME
+
+# Ensure app path directory exists and has appropriate file permissions.
+RUN mkdir -p $APP_PATH && \
+    chown $USER_ID:$USER_ID $APP_PATH && \
+    chmod g+s $APP_PATH
+
+# Update the system and install gettext for envsubst.
+RUN apk -U upgrade && \
+    apk add --update --no-cache gettext
 
 # Copy files from outside docker to inside.
 COPY build/appConfig.js.template /usr/local/app/templates/appConfig.js.template
 COPY build/docker-entrypoint.sh /usr/local/bin/docker-entrypoint.sh
 
-# Enable execute of docker entrypoint for root user.
-RUN chmod ugo+r /usr/local/app/templates/appConfig.js.template && \
-    chmod ugo+rx /usr/local/bin/docker-entrypoint.sh
-
-# Create the user and group (use a high ID to attempt to avoid conflicts).
-RUN groupadd --non-unique -g $USER_ID $USER_NAME && \
-    useradd --non-unique -d /$USER_NAME -m -u $USER_ID -g $USER_ID $USER_NAME
-
 # Login as user.
 USER $USER_NAME
 
 # Set deployment directory.
-WORKDIR /$USER_NAME
+WORKDIR $HOME_DIR
```

Comment:
It might be worth considering using `$HOME_DIR` rather than `/$USER_NAME` for othe docker files as well.

Reply:
Possibly. Did it startup on your development environment or are you still taking a piss?

Reply:
Haven't tested this yet.
I am merely remarking that we might want to apply this same change to Dockerfiles in our other repositories.